### ISSUE TITLE
Incorrectly reported warning CS1574: XML comment has cref attribute that could not be resolved

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#29070]
Repro: 

Checkout branch of https://github.com/dotnet/roslyn/pull/29069 (if not merged yet) and open

`src\VisualStudio\Core\Impl\CodeModel\CodeModelTaskExtensions.cs`

Change doc comment:

```xml
  /// <summary>
  /// Does a Roslyn.Utilities.TaskExtensions.WaitAndGetResult{T} for CodeModel.
  /// </summary>
```

to 

```xml
  /// <summary>
  /// Does a <see cref="Roslyn.Utilities.TaskExtensions.WaitAndGetResult{T}"/> for CodeModel.
  /// </summary>
```

Expected: 

No warning. 

Actual:

```
warning CS1574: XML comment has cref attribute 'WaitAndGetResult{T}' that could not be resolved
```

Calling method:
```C#
Roslyn.Utilities.TaskExtensions.WaitAndGetResult<T>(task, cancellationToken);
```

in the method body works, so the method is definitely accessible in this context.

**Comment by sharwell** [comment]
From my understanding of `cref` behavior, this is not a bug. I pointed out the offending line that caused this in #29069.

**Comment by tmat** [comment]
How do you explain that calling the method directly works? At least the error message needs to be improved if indeed its ambiguous.

**Comment by sharwell** [comment]
> How do you explain that calling the method directly works?

Only one of the definitions provides IVT to the call site. `cref` isn't bound by IVT restrictions.

**Comment by tmat** [comment]
I see.
